### ISSUE TITLE
Wrong golang version in hostedtoolcache

### ISSUE THREAD

**Issue by mokus** [actions/setup-go#207]
**Description:**
I'm using go-pdf/fpdf library in my project 
Using that library my project fails to build in github actions
The error is 

`[lambda:build:givematch:api] /home/runner/go/pkg/mod/github.com/go-pdf/fpdf@v0.6.0/embedded.go:22:2: package embed is not in GOROOT (/opt/hostedtoolcache/go/1.15.15/x64/src/embed)
`

I'm using golang v 1.17, so it's strange to see that it tries to load a module from 1.15.15

**Action version:**
actions/setup-go@v3

**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**
1.17

**Repro steps:**  
Using fpdf in a project

**Expected behavior:**
Project should build

**Actual behavior:**
Project fails to build

**Comment by smlx** [comment]
I'm seeing something similar. What was your fix here?

**Comment by IvanZosimov** [comment]
Hello, @smlx 👋 If you still have some troubles you can create a new issue with the detailed description and will try to help you ASAP 💪

**Comment by smlx** [comment]
Thanks for the offer. :slightly_smiling_face: 

In my case I was having a problem with https://github.com/jandelgado/gcov2lcov-action but I figured out that I needed to set `GOROOT` for [build.Import](https://pkg.go.dev/go/build#Import) to work correctly. In fact the `setup-go` action already populated the environment with the required info so I solved it like this:

```
- name: Set GOROOT
  run: |
    echo "GOROOT=$GOROOT_1_18_X64" >> $GITHUB_ENV
- name: Convert coverage to lcov
  uses: jandelgado/gcov2lcov-action@v1.0.8
```

I'll raise an issue over on that repo when I get a chance. I suspect it might be related to the very old version of Go that the latest binary of `gcov2lcov` was compled with.
